src/App.tsx

```diff
@@ -1,61 +1,138 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+
+export interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+export interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User | null;
+}
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const getNewTodoId = (newTodo: Todo[]): number => {
+  const maxId = Math.max(...newTodo.map((todo: Todo) => todo.id));
+
+  return maxId + 1;
+};
+
+const inTodo: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [isSelectUser, setIsSelectUser] = useState(false);
```

Comment:
The name `isSelectUser` but you set it to true when actually user isn't selected 🤷